### PR TITLE
feat(build): add support for building on arm and x86

### PR DESCRIPTION
**Pull request by wangzihao3**
Related to: openebs/openebs#1295
- modify build_image.sh to decide which arch is used
- add libkqueue deb which is not found in apt source.

Signed-off-by: wangzihao <wangzihao18@huawei.com>

### CODE REVIEW

File: docker/Dockerfile.arm64.base [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    apt-get update && apt-get install -y \
+    curl tcpdump dnsutils iputils-ping \
+    libaio1 libaio-dev \
+    libssl1.0.0 rsyslog net-tools gdb apt-utils \
+    sed libjemalloc-dev
+RUN apt-get -y install apt-file && apt-file update
+
+COPY libkqueue_2.3.1_arm64.deb /tmp/libkqueue_2.3.1_arm64.deb
+COPY libkqueue-dev_2.3.1_arm64.deb /tmp/libkqueue-dev_2.3.1_arm64.deb
+RUN dpkg -i /tmp/libkqueue_2.3.1_arm64.deb /tmp/libkqueue-dev_2.3.1_arm64.deb
```
Comment (mynktl) [inline docker/Dockerfile.arm64.base]: I didn't include this package in build in local compilation. Did you face any compilation issue without this package?

Comment (mynktl) [reply docker/Dockerfile.arm64.base]: This can be removed @Wangzihao18

Comment (wangzihao3) [reply docker/Dockerfile.arm64.base]: Hi @mynktl . No, there is no compilation issue without this package. I notice that the code using kqueue has been commented out, like istgt.c and istgt_iscsi.c. I will remove it now.

File: build_image.sh [hunk 2]
```diff
@@ -21,7 +24,18 @@ make clean
...
+if [ "${ARCH}" = "x86_64" ]; then
+	REPO_NAME="openebs/cstor-base"
+	DOCKERFILE_BASE="Dockerfile.base"
+	DOCKERFILE="Dockerfile"
+elif [ "${ARCH}" = "aarch64" ]; then
+	REPO_NAME="openebs/cstor-base-arm64"
+	DOCKERFILE_BASE="Dockerfile.arm64.base"
+	DOCKERFILE="Dockerfile.arm64"
+else
+	echo "Unusable architecture: ${ARCH}"
```
Comment (kmova) [inline build_image.sh]: Probably say -- "${ARCH} is not supported"

File: docker/Dockerfile.arm64.base [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#
```
Comment (kmova) [inline docker/Dockerfile.arm64.base]: Let this file follow the convention of Dockerfile.base.arm64

**Review by kmova — DISMISSED** [review]
Minor comments on text messages and file naming convention.

**Comment by wangzihao3** [comment]
Ok, @kmova . I will update these changes.

File: build_image.sh (hunk 2 above)
Comment (wangzihao3) [reply build_image.sh]: The message I followed this https://github.com/openebs/maya/blob/5ed267c117b7a5e293ec9d822db154782a9fdbc6/buildscripts/build.sh#L52 ^^.

**Review by mynktl — APPROVED** [review]

**Review by kmova — APPROVED** [review]
Thank you @Wangzihao18

**Comment by wangzihao3** [comment]
> Thank you @Wangzihao18

my pleasure!

**Comment by pawanpraka1** [comment]
nice changes @Wangzihao18. Thanks for this PR.

File: build_image.sh [hunk 4]
```diff
@@ -21,7 +24,18 @@ make clean
 make
 
 BUILD_DATE=$(date +'%Y%m%d%H%M%S')
-REPO_NAME="openebs/cstor-base"
+if [ "${ARCH}" = "x86_64" ]; then
+	REPO_NAME="openebs/cstor-base"
+	DOCKERFILE_BASE="Dockerfile.base"
+	DOCKERFILE="Dockerfile"
+elif [ "${ARCH}" = "aarch64" ]; then
+	REPO_NAME="openebs/cstor-base-arm64"
```
Comment (pawanpraka1) [inline build_image.sh:32]: we have to make sure that while building cstor-pool-mgmt image for arm64 it referes to this arm64 base image.

Comment (wangzihao3) [reply build_image.sh:32]: Thanks @pawanpraka1 . Yeah we should use different `CSTOR_BASE_IMAGE= openebs/cstor-base:${BASE_TAG}` when building for arm64. https://github.com/openebs/maya/blob/f18cf9f32d72c1f1bddfb89ad31743ef6d720caa/GNUmakefile#L96

**Pull request merged**